src/openforms/emails/signals.py

```diff
@@ -0,0 +1,29 @@
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from django_yubin.models import Message
+
+from openforms.logging import logevent
+from openforms.submissions.models import Submission
+
+from .constants import X_OF_CONTENT_UUID_HEADER, X_OF_EVENT_HEADER
+
+
+@receiver(post_save, sender=Message)
+def yubin_messages_status_change_handler(signal, sender, instance, created, **kwargs):
+    if created:
+        return
+
+    message = instance.get_email_message()
+    submission_uuid = message.extra_headers.get(X_OF_CONTENT_UUID_HEADER)
```

Comment:
wouldn't you need to check the content type header + value before knowing it's a submission-related email?

Reply:
Oh yes! Good point :smile: